src/os/ui/ogc/ogcserver.js

```diff
@@ -821,6 +843,64 @@ os.ui.ogc.OGCServer.prototype.displayConsentAlerts = function(result) {
 };
 
 
+/**
+ * Extract relevant information from the GetCapabilities result for display in Descriptors, HoverText, etc
+ * @param {string} constraints
+ * @return {?Array<string>}
+ */
+os.ui.ogc.OGCServer.prototype.getProperties = function(constraints) {
+  var properties = [];
+  var xmlDocument = null;
+
+  if (constraints && constraints.indexOf('xml version') && window.DOMParser) {
+    if (constraints.indexOf('&amp;lt;') >= 0) {
+      constraints = constraints.replace(/&amp;/g, '&');
+    }
+    try {
+      // clean up the HTML-encoded strings, e.g. &gt;
+      xmlDocument = (new DOMParser()).parseFromString(goog.string.unescapeEntities(constraints), 'text/xml');
+    } catch (e) {
+      goog.log.error(os.ui.ogc.OGCServer.LOGGER_, 'Could not parse XML for OGCServer additional information', e);
+    }
+  }
+
+  try {
+    var setting = /** @type {Object<?, osx.ogc.ServerProperty>} */ (os.config.Settings.getInstance()
```

Comment:
`?` should never be a valid key type for an object. Object keys are always strings (even if you try to make them numbers, they are strings).

Reply:
The compiler didn't like it when it was a string though... I'll see what other things I can do to trick it.